### PR TITLE
Glsk importer from farao new sensitivity API

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X ] The commit message follows our guidelines
- [X ] Tests for the changes have been added (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

No issue exists.

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

This is an adaptation of modules added in https://github.com/powsybl/powsybl-entsoe/pull/13 to the new sensitivity api developed in powsybl-core.

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

The main adaptation is the replacement of LinearGLSK by SensitivityVariableSet and WeightedSensitivityVariable.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [X ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by obrix** [comment]
@annetill  I'll do the modifications discussed in pull request 13 here.

File: glsk/glsk-document-api/src/main/java/com/powsybl/glsk/api/util/converters/GlskPointLinearGlskConverter.java [hunk 1]
```diff
@@ -35,9 +36,9 @@ private GlskPointLinearGlskConverter() {
      * @param glskPoint GLSK Point
      * @return farao-core LinearGlsk
      */
-    public static LinearGlsk convert(Network network, AbstractGlskPoint glskPoint) {
+    public static SensitivityVariableSet convert(Network network, AbstractGlskPoint glskPoint) {
 
-        Map<String, Float> linearGlskMap = new HashMap<>();
+        List<WeightedSensitivityVariable> weightedSVs = new ArrayList<>();
```
Comment (annetill) [inline glsk/glsk-document-api/src/main/java/com/powsybl/glsk/api/util/converters/GlskPointLinearGlskConverter.java:41]: we prefer in general an explicit name for variables. You can use here `weightedSensitivityVariables`.

File: glsk/glsk-document-cim/pom.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <!-- Dependencies versions -->
         <junit.version>4.13.1</junit.version>
-        <powsybl.core.version>4.4.0</powsybl.core.version>
+        <powsybl.core.version>4.6.0-SNAPSHOT</powsybl.core.version>
```
Comment (annetill) [inline glsk/glsk-document-cim/pom.xml:23]: I don't understand why you have to rewrite the version everywhere. Can you look at this other repositories to see how this is managed efficiently?

File: glsk/glsk-document-cse/pom.xml [hunk 3]
```diff
@@ -20,7 +20,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <!-- Dependencies versions -->
         <junit.version>4.13.1</junit.version>
-        <powsybl.core.version>4.4.0</powsybl.core.version>
+        <powsybl.core.version>4.6.0-SNAPSHOT</powsybl.core.version>
```
Comment (annetill) [inline glsk/glsk-document-cse/pom.xml:23]: Same comment.

File: glsk/glsk-document-ucte/pom.xml [hunk 4]
```diff
@@ -22,7 +22,7 @@
         <jackson.version>2.11.2</jackson.version>
         <jimfs.version>1.1</jimfs.version>
         <junit.version>4.13.1</junit.version>
-        <powsybl.core.version>4.4.0</powsybl.core.version>
+        <powsybl.core.version>4.6.0-SNAPSHOT</powsybl.core.version>
```
Comment (annetill) [inline glsk/glsk-document-ucte/pom.xml:25]: Same comment.

File: glsk/glsk-document-ucte/src/test/java/com/powsybl/glsk/ucte/UcteGlskValueProviderTest.java [hunk 5]
```diff
@@ -179,7 +179,12 @@ public void testZeroLsk() {
         Network network = Importers.loadNetwork("testCase.xiidm", getClass().getResourceAsStream("/testCase.xiidm"));
         Instant instant = Instant.parse("2016-07-29T10:00:00Z");
         UcteGlskDocument ucteGlskDocument = UcteGlskDocument.importGlsk(getClass().getResourceAsStream("/TestZeroLsk.xml"));
-        ZonalData<LinearGlsk> ucteGlskProvider = ucteGlskDocument.getZonalGlsks(network, instant);
+        ZonalData<SensitivityVariableSet> ucteGlskProvider = ucteGlskDocument.getZonalGlsks(network, instant);
         assertNull(ucteGlskProvider.getData("10YFR-RTE------C"));
     }
+
+    //For readability purpose
+    private double findWeight(SensitivityVariableSet variables, String id) {
```
Comment (annetill) [inline glsk/glsk-document-ucte/src/test/java/com/powsybl/glsk/ucte/UcteGlskValueProviderTest.java:187]: Maybe it should be interesting to put this method in the sensitivity API?

Comment (obrix) [reply glsk/glsk-document-ucte/src/test/java/com/powsybl/glsk/ucte/UcteGlskValueProviderTest.java:187]: Yes but not this way. This is an inefficient way to go around the fact the SensitivityVariables are stored as a list and not indexed by id. We can discuss the idea to modify the api.

File: glsk/glsk-document-cim/pom.xml (hunk 2 above)
Comment (obrix) [reply glsk/glsk-document-cim/pom.xml:23]: Actually there is mismatch on the property name between the poms : powsyblcore on the parent -> powsybl.core on the children. That's why the property was not correctly propagated. Fixing it.

File: glsk/glsk-document-cse/pom.xml (hunk 3 above)
Comment (obrix) [reply glsk/glsk-document-cse/pom.xml:23]: idem

File: glsk/glsk-document-ucte/pom.xml (hunk 4 above)
Comment (obrix) [reply glsk/glsk-document-ucte/pom.xml:25]: idem

**Pull request closed without merging**